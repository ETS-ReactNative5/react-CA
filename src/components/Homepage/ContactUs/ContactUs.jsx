import React, { Component } from 'react'
import { NavLink } from 'react-router-dom'

import Button from 'components/Core/Button/Button'
import FormGroup from 'components/Core/FormGroup/FormGroup'
import ContentPage from 'components/Core/ContentPage/ContentPage'
import { Row, Col } from 'components/Core/Section/Section'
import ProgramModal from 'components/Core/ProgramModal/ProgramModal'
import Footer from 'containers/Footer/Footer'

import { commonService } from 'utils/api/common'
import { validators } from 'utils/formatters/validators'
import { log } from 'utils/log'

import mailIcon from 'assets/images/homepage/mail.svg'
import msgIcon from 'assets/images/homepage/pencil.svg'
import styles from './ContactUs.scss'


class ContactUs extends Component {
  state = {
    email: '',
    message: '',
    formValid: false,
    isConfirm: false,
    showDisclaimerModal: false,
  }

  getContactFormInputs = () => {
    const { globals } = validators
    const { email, message } = this.state

    return [
      { id: 'email', value: email, validators: globals.email, handler: value => this.handleStoreFormChange('email', value), placeholder: 'Email', icon: mailIcon },
      { id: 'message', value: message, handler: value => this.handleStoreFormChange('message', value), placeholder: 'Your message', icon: msgIcon },
    ]
  }

  handleToggleModal = () => (
    this.setState(ps => ({ ...ps, showDisclaimerModal: !ps.showDisclaimerModal }))
  )

  handleStoreFormChange = (stateKey, value) => (
    this.setState(ps => ({ ...ps, [stateKey]: value }))
  )

  handleOnSubmit = () => {
    const { email, phone, name, message } = this.state
    const payload = { email, phone, name, message }

    commonService
      .contactUs(payload)
      .then(this.handleSuccess)
      .catch(this.handleError)
  }

  handleSuccess = (response) => {
    log.info(response)
    this.setState(ps => ({ ...ps,
      email: '',
      message: '',
      formValid: false,
      isConfirm: true,
    }))
  }

  handleError = response => log.error(response)

  render() {
    const { formValid, isConfirm, showDisclaimerModal } = this.state

    return (
      <div className={styles.ContactUs}>
        {isConfirm ?
          <ContentPage title='Confirmation' blue>
            <Row>
              <Col md='8'>
                <p className={styles.conformMessage}>
                  {
                    `Thank you! We’ve received your message and we’ll get
                    back to you shortly. Credit Academy`
                  }
                </p>
                <NavLink to='/'>
                  <Button primary text='Return to Homepage' />
                </NavLink>
              </Col>
            </Row>
          </ContentPage>
        :
          <ContentPage title='Contact Us' blue>
            <Row>
              <Col md='8'>
                <FormGroup
                  inputs={this.getContactFormInputs()}
                  isValid={value => this.handleStoreFormChange('formValid', value)}
                />
                <Button
                  className={styles.button}
                  onClick={this.handleOnSubmit}
                  disabled={!formValid}
                  text='Send'
                  primary
                />
              </Col>
            </Row>
          </ContentPage>
        }
        { <Footer handleToggleModal={this.handleToggleModal} /> }
        { showDisclaimerModal && <ProgramModal handleToggleModal={this.handleToggleModal} />}
      </div>
    )
  }
}

ContactUs.propTypes = {}
ContactUs.defaultProps = {}

export default ContactUs
