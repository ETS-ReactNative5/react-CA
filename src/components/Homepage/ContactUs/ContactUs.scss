.ContactUs {
  .conformMessage {
    color: white;
    padding: 10px 0;
  }
}
