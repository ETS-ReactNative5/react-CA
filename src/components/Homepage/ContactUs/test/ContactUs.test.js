import React from 'react'
import { shallow } from 'enzyme'

import ContactUs from '../ContactUs'


describe('<ContactUs />', () => {
  it('renders correctly', () => {
    const wrapper = shallow(<ContactUs />)
    expect(wrapper).toMatchSnapshot()
  })
})
