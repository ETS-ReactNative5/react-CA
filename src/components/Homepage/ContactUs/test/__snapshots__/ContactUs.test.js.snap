// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ContactUs /> renders correctly 1`] = `
<div
  className="ContactUs"
>
  <ContentPage
    blue={true}
    title="Contact Us"
  >
    <Row
      className=""
    >
      <Col
        className=""
        lg={null}
        lg-offset={null}
        md="8"
        md-offset={null}
        sm={null}
        sm-offset={null}
        xl={null}
        xl-offset={null}
        xs={null}
        xs-offset={null}
      >
        <FormGroup
          className={null}
          inputs={
            Array [
              Object {
                "handler": [Function],
                "icon": "mail.svg",
                "id": "email",
                "placeholder": "Email",
                "validators": Array [
                  Object {
                    "error": Object {
                      "level": 3,
                      "message": "Please enter a valid email address!",
                    },
                    "regex": /\\^\\[a-zA-Z0-9\\.\\\\-_\\]\\+@\\[a-zA-Z0-9\\\\-_\\]\\+\\\\\\.\\[a-zA-Z0-9\\.\\\\-_\\]\\+\\$/,
                  },
                ],
                "value": "",
              },
              Object {
                "handler": [Function],
                "icon": "pencil.svg",
                "id": "message",
                "placeholder": "Your message",
                "value": "",
              },
            ]
          }
          isValid={[Function]}
        />
        <Button
          className="button"
          disabled={true}
          icon={null}
          iconOnly={false}
          middle={true}
          nocaps={false}
          onClick={[Function]}
          outline={false}
          primary={true}
          secondary={false}
          small={false}
          text="Send"
          textOnly={false}
          tiny={false}
        />
      </Col>
    </Row>
  </ContentPage>
  <Responsive
    handleToggleModal={[Function]}
  />
</div>
`;
